pallets/issue/src/tests.rs

```diff
@@ -624,3 +624,643 @@ fn test_request_issue_reset_interval_and_succeeds_with_rate_limit() {
+			//set up the vaults
+			let collateral_1: u128 = 1000;
+			register_vault_with_collateral(&VAULT, collateral_1);
+
+			let collateral_2: u128 = 2000;
+			register_vault_with_collateral(&VAULT_2, collateral_2);
+
+			let collateral_3: u128 = 1500;
+			register_vault_with_collateral(&VAULT_3, collateral_3);
+
+			let collateral_4: u128 = 1700;
+			register_vault_with_collateral(&VAULT_4, collateral_4);
+
+			let collateral_5: u128 = 800;
+			register_vault_with_collateral(&VAULT_5, collateral_5);
+
+			//execute issues
+			let issue_asset_1 = VAULT.wrapped_currency();
+			let issue_asset_2 = VAULT_2.wrapped_currency();
+			let issue_asset_3 = VAULT_3.wrapped_currency();
+
+			//ACT
+			//get issue total for asset1, asset2 and asset3
+			let (issue_fee_1, issue_fee_2, issue_fee_3) = execute_multiple_issues();
+
+			//ASSERT
+			//ensure that the current rewards equal to fee
+			//Example: we expect for reward(vault1, asset1) = floor(
+			//floor((1000/3000)*floor( 1400*(3000*5/(3000*5+3200*10+800*15)) )) = 118
+
+			let vault_1_collateral_usd = to_usd(&collateral_1, &VAULT.collateral_currency());
+			let vault_2_collateral_usd = to_usd(&collateral_2, &VAULT_2.collateral_currency());
+			let vault_3_collateral_usd = to_usd(&collateral_3, &VAULT_3.collateral_currency());
+			let vault_4_collateral_usd = to_usd(&collateral_4, &VAULT_4.collateral_currency());
+			let vault_5_collateral_usd = to_usd(&collateral_5, &VAULT_5.collateral_currency());
+
+			let total_amount_usd = vault_1_collateral_usd +
+				vault_2_collateral_usd +
+				vault_3_collateral_usd +
+				vault_4_collateral_usd +
+				vault_5_collateral_usd;
+
+			let get_expected_value_vault_1 = |fee: Balance| -> Balance {
+				((((vault_1_collateral_usd + vault_2_collateral_usd) as f64 /
+					total_amount_usd as f64) *
+					fee as f64)
+					.floor() * (collateral_1 as f64 / (collateral_2 + collateral_1) as f64))
+					.floor() as u128
+			};
+
+			let get_expected_value_vault_2 = |fee: Balance| -> Balance {
+				((((vault_1_collateral_usd + vault_2_collateral_usd) as f64 /
+					total_amount_usd as f64) *
+					fee as f64)
+					.floor() * (collateral_2 as f64 / (collateral_2 + collateral_1) as f64))
+					.floor() as u128
+			};
+
+			let get_expected_value_vault_3 = |fee: Balance| -> Balance {
+				((((vault_3_collateral_usd + vault_4_collateral_usd) as f64 /
+					total_amount_usd as f64) *
+					fee as f64)
+					.floor() * (collateral_3 as f64 / (collateral_3 + collateral_4) as f64))
+					.floor() as u128
+			};
+
+			let get_expected_value_vault_4 = |fee: Balance| -> Balance {
+				((((vault_3_collateral_usd + vault_4_collateral_usd) as f64 /
+					total_amount_usd as f64) *
+					fee as f64)
+					.floor() * (collateral_4 as f64 / (collateral_3 + collateral_4) as f64))
+					.floor() as u128
+			};
+
+			let get_expected_value_vault_5 = |fee: Balance| -> Balance {
+				(((vault_5_collateral_usd as f64 / total_amount_usd as f64) * fee as f64).floor())
+					as u128
+			};
```

Comment:
Here it is not clear at first, why vault 1+2, vault 3+4 and vault 5 are grouped together.
Again, looking at the mocks you can see that these vaults share the same collateral currency. But as a reader I wouldn't understand this from just looking at the test. Let's maybe add a comment above the getter functions that explains why some vault collaterals are grouped together.